ui/imodel-components/src/ui-imodel-components/navigationaids/DrawingNavigationAid.tsx

```diff
@@ -763,7 +763,8 @@ export class DrawingNavigationCanvas extends React.Component<DrawingNavigationCa
     /* istanbul ignore next */
     if (vp.logo.style)
       vp.logo.style.display = "none";
-    vp.viewFlags.acsTriad = false;
+
+    vp.viewFlags = vp.viewFlags.with("acsTriad", false);

```

Comment:
@bsteinbk @DanEastBentley can one of you tell me how to satisfy moq for DrawingNavigationAid.test.tsx?
With no changes to the test code, I get 8 failures: "vp.viewFlags.with is not a function".
If I add `vpMock.setup((x) => x.viewFlags).returns(() => new ViewFlags())` I get 8 failures: "trap returned falsish for property".

Reply:
You'll probably need something like:
```ts
const viewFlagsMock = moq.Mock.ofType<ViewFlags>();
viewFlagsMock.setup((vf) => vf.with).returns(() => . . .);  // You'll need to provide implementation
. . .
vp.setup((x) => x.viewFlags).returns(() => viewFlagsMock.object);
```

Reply:
```
      const viewFlags = new ViewFlags();
      const viewFlagsMock = moq.Mock.ofType<ViewFlags>();
      viewFlagsMock.setup((vf) => vf.with).returns((_vf) => (prop, value) => {
        return viewFlags.with(prop, value);
      });

      const vpMock = moq.Mock.ofType<ScreenViewport>();
      vpMock.setup((x) => x.view).returns(() => viewState.object);
      vpMock.setup((x) => x.viewFlags).returns(() => viewFlagsMock.object);
      return vpMock.object;
```
"trap returned falsish for property".

Reply:
I am not sure why Paul's solution did not work unless we need to return the same set of flags each time.
```
const testViewFlags = new ViewFlags();
vp.setup((x) => x.viewFlags).returns(() => testViewFlags );
```

Reply:
Tried that one already, @bsteinbk - same "'set' on proxy: trap returned falsish for property 'viewFlags'" error.

Reply:
How large is the ViewFlag.with implementation? Can you include it in the test instead of relying on a real ViewFlags object?

Reply:
It's small, but for this test it could be replaced with a mock that simply returns the same object, or a new one, or whatever - the test shouldn't care.
The problem though appears to be the assignment to `Viewport.viewFlags`, not the call of `with`.
```
  public with(flag: keyof Omit<ViewFlagsProperties, "renderMode">, value: boolean): ViewFlags {
    if (this[flag] === value)
      return this;

    const props: ViewFlagsProperties = { ...this };
    props[flag] = value;
    return new ViewFlags(props);
  }
```

Reply:
I am not sure why there are two viewport mocks in these tests,  one named `vp` and one `ScreenViewportMock`.  Did you add viewFlags to both? 

`vp.setup((x) => x.viewFlags).returns(() => testViewFlags );`
and 
`vpMock.setup((x) => x.viewFlags).returns(() => testViewFlags);`

Reply:
I didn't notice that. But the callstack shows the errors happening in this mock, not that other one.
Adding viewFlags to the other one made no difference.

Reply:
I gave up trying to figure out if this is a me problem or a typemoq problem, and just changed customizeViewport to only assign to viewFlags if necessary.
For the test, it is not necessary.

Reply:
You will need to add /* istanbul ignore next */ above your new if statement or the test coverage will not be 100% and cause you more grief.